lvsfunc.py

```diff
@@ -29,132 +56,169 @@ def compare(clip_a: vs.VideoNode, clip_b: vs.VideoNode, frames: int, mark=False,
 
-    If Eedi3=False, it will use Nnedi3 instead.
     """
-    srcY = vsutil.get_y(clip)
+    src_y = get_y(clip)
     height, width = clip.height, clip.width
 
     if eedi3:
-        def aa(srcY):
-            srcY = srcY.std.Transpose()
-            srcY = srcY.eedi3m.EEDI3(0, 1, 0, 0.5, 0.2)
-            srcY = srcY.znedi3.nnedi3(1, 0, 0, 3, 4, 2)
-            srcY = srcY.resize.Spline36(height, width, src_top=.5)
-            srcY = srcY.std.Transpose()
-            srcY = srcY.eedi3m.EEDI3(0, 1, 0, 0.5, 0.2)
-            srcY = srcY.znedi3.nnedi3(1, 0, 0, 3, 4, 2)
-            srcY = srcY.resize.Spline36(width, height, src_top=.5)
-            return srcY
+        def _aa(src_y):
+            src_y = src_y.std.Transpose()
+            src_y = src_y.eedi3m.EEDI3(0, 1, 0, 0.5, 0.2)
+            src_y = src_y.znedi3.nnedi3(1, 0, 0, 3, 4, 2)
+            src_y = src_y.resize.Spline36(height, width, src_top=.5)
+            src_y = src_y.std.Transpose()
+            src_y = src_y.eedi3m.EEDI3(0, 1, 0, 0.5, 0.2)
+            src_y = src_y.znedi3.nnedi3(1, 0, 0, 3, 4, 2)
+            src_y = src_y.resize.Spline36(width, height, src_top=.5)
+            return src_y
     else:
-        def aa(srcY):
-            srcY = srcY.std.Transpose()
-            srcY = srcY.nnedi3.nnedi3(0, 1, 0, 3, 3, 2)
-            srcY = srcY.nnedi3.nnedi3(1, 0, 0, 3, 3, 2)
-            srcY = srcY.resize.Spline36(height, width, src_top=.5)
-            srcY = srcY.std.Transpose()
-            srcY = srcY.nnedi3.nnedi3(0, 1, 0, 3, 3, 2)
-            srcY = srcY.nnedi3.nnedi3(1, 0, 0, 3, 3, 2)
-            srcY = srcY.resize.Spline36(width, height, src_top=.5)
-            return srcY
-
-    def csharp(flt, src):
+        def _aa(src_y):
+            src_y = src_y.std.Transpose()
+            src_y = src_y.nnedi3.nnedi3(0, 1, 0, 3, 3, 2)
+            src_y = src_y.nnedi3.nnedi3(1, 0, 0, 3, 3, 2)
+            src_y = src_y.resize.Spline36(height, width, src_top=.5)
+            src_y = src_y.std.Transpose()
+            src_y = src_y.nnedi3.nnedi3(0, 1, 0, 3, 3, 2)
+            src_y = src_y.nnedi3.nnedi3(1, 0, 0, 3, 3, 2)
+            src_y = src_y.resize.Spline36(width, height, src_top=.5)
+            return src_y
+
+    def _csharp(flt, src):
         blur = core.std.Convolution(flt, [1] * 9)
         return core.std.Expr([flt, src, blur], 'x y < x x + z - x max y min x x + z - x min y max ?')
 
-    aaclip = aa(srcY)
-    aaclip = csharp(aaclip, srcY).rgvs.Repair(srcY, 13)
+    aaclip = _aa(src_y)
+    aaclip = _csharp(aaclip, src_y).rgvs.Repair(src_y, 13)
 
     return aaclip if clip.format.color_family is vs.GRAY else core.std.ShufflePlanes([aaclip, clip], [0, 1, 2], vs.YUV)
 
 
-def NnEedi3(clip: vs.VideoNode, mask=None, strong_mask=False, show_mask=False, opencl=False, strength=1, alpha=0.25, beta=0.5, gamma=40, nrad=2, mdis=20, nsize=3, nns=3, qual=1):
+def nneedi3_clamp(clip: vs.VideoNode, mask=None, strong_mask: bool = False, show_mask: bool = False,
```

Comment:
maybe we should consider `**kwargs` here for one of the AA calls. But that’s beyond the scope of this commit.